mlrun/api/db/sqldb/models/models_mysql.py

```diff
@@ -139,6 +139,8 @@ class Notification(Base, mlrun.utils.db.BaseModel):
 with warnings.catch_warnings():
     warnings.simplefilter("ignore")
 
+    # deprecated, use ArtifactV2 instead
+    # TODO: remove in 1.6.0
```

Comment:
Note that removing the table in 1.6.0 means that migrations might not work between 1.4.0(or lower) -> 1.6.0.
And you will have to do multi stage upgrade 1.4.0 -> 1.5.0 and then 1.5.0 -> 1.6.0 ( as only 1.5.0 will be familiar with the `Artifact` table )

Reply:
also when we deprecate objects / methods in MLRun it contains few factors
1. whether it is user facing or not ( accessible through the SDK )
2. whether it is API ( older clients can access it )
3. How would it effect upgrades

our rule of thumb when one of the above matches is removing the object 2 versions aheads.
so either way I would suggest setting the target to remove to 1.7.0, and also we might decide to carry with it a bit longer depends on what our clients will be upgrading from.